File: src/test/java/burrows/apps/example/gif/presentation/ImmediateSchedulerProvider.java
package burrows.apps.example.gif.presentation;

import io.reactivex.Scheduler;
import io.reactivex.schedulers.Schedulers;

/**
 * @author <a href="mailto:jaredsburrows@gmail.com">Jared Burrows</a>
 */
public final class ImmediateSchedulerProvider implements IBaseSchedulerProvider {
  @Override public Scheduler io() {
    return Schedulers.trampoline();
  }

  @Override public Scheduler ui() {
    return Schedulers.trampoline();
  }
}

File: src/test/java/test/ImmediateSchedulerProvider.kt
package test

import burrows.apps.example.gif.SchedulerProvider
import io.reactivex.schedulers.Schedulers
import javax.inject.Inject

class ImmediateSchedulerProvider @Inject constructor() : SchedulerProvider() {
  override fun io() = Schedulers.trampoline()
  override fun ui() = Schedulers.trampoline()
}
